torch/nn/parallel/distributed.py

```diff
@@ -305,14 +305,16 @@ class DistributedDataParallel(Module):
 
     Args:
         module (Module): module to be parallelized
-        device_ids (list of int or torch.device): CUDA devices. This should
-                   only be provided when the input module resides on a single
-                   CUDA device. For single-device modules, the i'th
-                   :attr:`module` replica is placed on ``device_ids[i]``. For
-                   multi-device modules and CPU modules, ``device_ids`` must be
-                   ``None`` or an empty list, and input data for the forward
-                   pass must be placed on the correct device. (default: all
-                   visible devices for single-device modules)
+        device_ids (list of int or torch.device): CUDA devices.
```

Comment:
I wonder just for semantics, if it's worth providing a new `device` argument, so that new users don't have to wrap their head around the awkwardness of specifying only one device in a list of `device_ids` (though admittedly it's not much of an issue). Although, I'm guessing this will make the API too verbose to be feasible?

Reply:
I think the existing API should be fine, as long as we view SPMD as a special case of a multi-device module, where the # of devices in a module is 1.

If we provide both `device` and `device_ids`, do you mean we shouldn't allow `device_ids` to contain a single element? If allowed, then the user can have two routes to control the same config -- what if the single id in `device` field and `device_ids` differ? This may also bring some unnecessary confusion.